### PR TITLE
Standardize references to the UTF-8 term in code and comments/documentation

### LINKED ISSUE #61846: New public APIs for Checked User-defined Operators
```diff
namespace Microsoft.CodeAnalysis
{
    public static class WellKnownMemberNames
    {
+        /// <summary>
+        /// The name assigned to a chacked explicit (narrowing) conversion.
+        /// </summary>
+        public const string CheckedExplicitConversionName = "op_CheckedExplicit";

+        /// <summary>
+        /// The name assigned to the checked Addition operator.
+        /// </summary>
+        public const string CheckedAdditionOperatorName = "op_CheckedAddition";

+        /// <summary>
+        /// The name assigned to the checked Decrement operator.
+        /// </summary>
+        public const string CheckedDecrementOperatorName = "op_CheckedDecrement";

+        /// <summary>
+        /// The name assigned to the checked Division operator.
+        /// </summary>
+        public const string CheckedDivisionOperatorName = "op_CheckedDivision";

+        /// <summary>
+        /// The name assigned to the checked Increment operator.
+        /// </summary>
+        public const string CheckedIncrementOperatorName = "op_CheckedIncrement";

+        /// <summary>
+        /// The name assigned to the checked Multiply operator.
+        /// </summary>
+        public const string CheckedMultiplyOperatorName = "op_CheckedMultiply";

+        /// <summary>
+        /// The name assigned to the checked Subtraction operator.
+        /// </summary>
+        public const string CheckedSubtractionOperatorName = "op_CheckedSubtraction";

+        /// <summary>
+        /// The name assigned to the checked UnaryNegation operator.
+        /// </summary>
+        public const string CheckedUnaryNegationOperatorName = "op_CheckedUnaryNegation";
    }
}
```

### LINKED ISSUE #61847: New public APIs for UTF8 String Literals
```diff
namespace Microsoft.CodeAnalysis
{
    public enum OperationKind
    {
+        /// <summary>Indicates an <see cref="IUTF8StringOperation"/>.</summary>
+        UTF8String = 0x7c,
    }
}

namespace Microsoft.CodeAnalysis.Operations
{
+    /// <summary>
+    /// Represents a UTF8 encoded byte representation of a string.
+    /// <para>
+    ///   Current usage:
+    ///   (1) C# UTF8 string literal expression.
+    /// </para>
+    /// </summary>
+    /// <remarks>
+    /// <para>This node is associated with the following operation kinds:</para>
+    /// <list type="bullet">
+    /// <item><description><see cref="OperationKind.UTF8String"/></description></item>
+    /// </list>
+    /// <para>This interface is reserved for implementation by its associated APIs. We reserve the right to
+    /// change it in the future.</para>
+    /// </remarks>
+    public interface IUTF8StringOperation : IOperation
+    {
+        /// <summary>
+        /// The underlying string value.
+        /// </summary>
+        string Value { get; }
+    }

    public abstract partial class OperationVisitor
    {
+        public virtual void VisitUTF8String(IUTF8StringOperation operation) => DefaultVisit(operation);
    }

    public abstract partial class OperationVisitor<TArgument, TResult>
    {
+        public virtual TResult? VisitUTF8String(IUTF8StringOperation operation, TArgument argument) => DefaultVisit(operation, argument);
    }
}

+Microsoft.CodeAnalysis.CSharp.SyntaxKind.UTF8StringLiteralExpression = 8756 -> Microsoft.CodeAnalysis.CSharp.SyntaxKind
+Microsoft.CodeAnalysis.CSharp.SyntaxKind.UTF8StringLiteralToken = 8520 -> Microsoft.CodeAnalysis.CSharp.SyntaxKind
+Microsoft.CodeAnalysis.CSharp.SyntaxKind.UTF8SingleLineRawStringLiteralToken = 8521 -> Microsoft.CodeAnalysis.CSharp.SyntaxKind
+Microsoft.CodeAnalysis.CSharp.SyntaxKind.UTF8MultiLineRawStringLiteralToken = 8522 -> Microsoft.CodeAnalysis.CSharp.SyntaxKind
```

### PR DESCRIPTION
**Pull request by AlekseyTs**
From a framework naming perspective, “Utf8” is preferred, because it’s a 3 letter abbreviation (for Unicode Transformation Format), and 3+ letter abbreviations should be PascalCased, like Html and Uri.

From a documentation/conceptual perspective, the preferred spelling of the Unicode Consortium seems to be “UTF-8” (all caps, with a hyphen).

Closes #61847.

Also fixes a typo and closes #61846.

### CODE REVIEW

**Review by jcouv — APPROVED** [review]
LGTM Thanks, both compiler and IDE changes looked at (iteration 5)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review.

**Review by davidwengier — APPROVED** [review]
IDE looks good

**Comment by AlekseyTs** [comment]
@cston, @RikkiGibson, @dotnet/roslyn-compiler For the second review.

**Comment by AlekseyTs** [comment]
@davidwengier Please take a look at IDE side of changes.

File: src/Compilers/Core/Portable/Serialization/ObjectWriter.cs [hunk 1]
```diff
@@ -1307,8 +1307,8 @@ internal enum EncodingKind : byte
             EncodingName,
 
             // well-known encodings (parameterized by BOM)
-            EncodingUTF8,
-            EncodingUTF8_BOM,
+            EncodingUtf8,
+            EncodingUtf8_BOM,
```
Comment (cston) [inline src/Compilers/Core/Portable/Serialization/ObjectWriter.cs:1311]: It looks like these should be kept as-is, or "EncodingUTF32*" should be updated as well.

**Pull request merged**